pkg/agent/controller/networkpolicy/packetin.go

```diff
@@ -327,3 +333,81 @@ func (c *Controller) rejectRequest(pktIn *ofctrl.PacketIn) error {
 			true)
 	}
 }
+
+func (c *Controller) storeDenyConnection(pktIn *ofctrl.PacketIn) error {
+	packet, err := binding.ParsePacketIn(pktIn)
+	if err != nil {
+		return fmt.Errorf("error in parsing packetin: %v", err)
+	}
+
+	// Get 5-tuple information
+	tuple := flowexporter.Tuple{
+		SourcePort:      packet.SourcePort,
+		DestinationPort: packet.DestinationPort,
+		Protocol:        packet.IPProto,
+	}
+	if packet.IsIPv6 {
+		tuple.SourceAddress = packet.SourceIP.To16()
+		tuple.DestinationAddress = packet.DestinationIP.To16()
+	} else {
+		tuple.SourceAddress = packet.SourceIP.To4()
+		tuple.DestinationAddress = packet.DestinationIP.To4()
+	}
+
+	// Generate deny connection and add to deny connection store
+	denyConn := flowexporter.Connection{}
+	denyConn.FlowKey = tuple
+	denyConn.DestinationServiceAddress = tuple.DestinationAddress
+	denyConn.DestinationServicePort = tuple.DestinationPort
+
+	// No need to obtain connection info again if it already exists in denyConnectionStore.
+	if conn, exist := c.denyConnStore.GetConnByKey(flowexporter.NewConnectionKey(&denyConn)); exist {
+		c.denyConnStore.AddOrUpdateConn(conn, time.Now(), uint64(packet.IPLength))
+		return nil
+	}
+
+	matchers := pktIn.GetMatches()
+	var match *ofctrl.MatchField
+	// Get table ID
+	tableID := binding.TableIDType(pktIn.TableId)
+	// Get disposition Allow, Drop or Reject
+	match = getMatchRegField(matchers, uint32(openflow.DispositionMarkReg))
+	id, err := getInfoInReg(match, openflow.APDispositionMarkRange.ToNXRange())
+	if err != nil {
+		return fmt.Errorf("error when getting disposition from reg: %v", err)
+	}
+	disposition := openflow.DispositionToString[id]
+
+	// For K8s NetworkPolicy implicit drop action, we cannot get name/namespace.
+	if tableID == openflow.IngressDefaultTable {
+		denyConn.IngressNetworkPolicyRuleAction = flowexporter.RuleActionToUint8(disposition)
+	} else if tableID == openflow.EgressDefaultTable {
+		denyConn.EgressNetworkPolicyRuleAction = flowexporter.RuleActionToUint8(disposition)
+	} else { // Get name and namespace for Antrea Network Policy or Antrea Cluster Network Policy
+		// Set match to corresponding ingress/egress reg according to disposition
+		match = getMatch(matchers, tableID, id)
+		ruleID, err := getInfoInReg(match, nil)
+		if err != nil {
+			return fmt.Errorf("error when obtaining rule id from reg: %v", err)
+		}
+		policy := c.GetNetworkPolicyByRuleFlowID(ruleID)
+
+		if policy == nil {
+			// Default drop by k8s network policy
+			klog.V(2).Infof("Cannot find NetworkPolicy that has ruleID %v", ruleID)
```

Comment:
```suggestion
			klog.V(4).Infof("Cannot find NetworkPolicy that has ruleID %v", ruleID)
```

decreased verbosity a bit since this is normal operations and probably quite frequent